
Hotfix from X.Org Bugzilla to prevent X from haning during startup
on e.g. PowerPC Apple {i,Power}Books ...

  - Rene Rebe <rene@rocklinux.org>

Index: programs/Xserver/hw/xfree86/drivers/ati/radeon_driver.c
===================================================================
RCS file: /cvs/xorg/xc/programs/Xserver/hw/xfree86/drivers/ati/radeon_driver.c,v
retrieving revision 1.16
diff -u -r1.16 radeon_driver.c
--- xc/programs/Xserver/hw/xfree86/drivers/ati/radeon_driver.c	10 Aug 2004 04:01:42 -0000	1.16
+++ xc/programs/Xserver/hw/xfree86/drivers/ati/radeon_driver.c	11 Aug 2004 18:11:55 -0000
@@ -1362,8 +1362,8 @@
                     info->Flags = 0;
                     break;
                 }
-                tmp_mode = tmp_mode->next;
             }
+            tmp_mode = tmp_mode->next;
         }
         if ((info->DotClock == 0) && !pRADEONEnt->PortInfo[0].MonInfo) {
             xf86DrvMsg(pScrn->scrnIndex, X_ERROR,
